src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -105,21 +109,38 @@ public async Task ExecuteAsync(ReindexJobRecord reindexJobRecord, WeakETag weakE
                     var resourceList = new HashSet<string>();
                     foreach (var param in notYetIndexedParams)
                     {
-                        if (param.TargetResourceTypes != null)
+                        foreach (var baseResourceType in param.BaseResourceTypes)
                         {
-                            resourceList.UnionWith(param.TargetResourceTypes);
-                        }
+                            if (baseResourceType == "Resource")
+                            {
+                                resourceList.UnionWith(_modelInfoProvider.GetResourceTypeNames().ToHashSet());
 
-                        if (param.BaseResourceTypes != null)
-                        {
-                            resourceList.UnionWith(param.BaseResourceTypes);
+                                // We added all possible resource types, so no need to continue
+                                break;
+                            }
+
+                            if (baseResourceType == "DomainResource")
+                            {
+                                var domainResourceChildResourceTypes = _modelInfoProvider.GetResourceTypeNames().ToHashSet();
+
+                                // Remove types that inherit from Resource directly
+                                domainResourceChildResourceTypes.Remove("Binary");
+                                domainResourceChildResourceTypes.Remove("Bundle");
+                                domainResourceChildResourceTypes.Remove("Parameters");
```

Comment:
nit: KnownResources.Parameters

Reply:
That's much nicer, thanks. Fixed in a80b86e.